src/Zio/FileSystems/MountFileSystem.cs

```diff
@@ -511,115 +526,135 @@ protected override IEnumerable<UPath> EnumeratePathsImpl(UPath path, string sear
         {
             // Use the search pattern to normalize the path/search pattern
             var search = SearchPattern.Parse(ref path, ref searchPattern);
-            var originalSrcPath = path;
 
-            // Internal method used to retrieve the list of root directories
-            SortedSet<UPath> GetRootDirectories()
+            // Internal method used to retrieve the list of search locations
+            IEnumerable<SearchLocation> GetSearchLocations(UPath basePath)
             {
-                var directories = new SortedSet<UPath>(UPath.DefaultComparerIgnoreCase);
                 lock (_mounts)
                 {
-                    foreach (var mountName in _mounts.Keys)
+                    foreach (var kvp in _mounts)
                     {
-                        directories.Add(mountName);
-                    }
-                }
+                        // Check if path partially matches a mount name
+                        var remainingPath = GetRemaining(basePath, kvp.Key);
+                        if (!remainingPath.IsNull && remainingPath != UPath.Root)
+                        {
+                            yield return new SearchLocation(this, basePath, remainingPath);
```

Comment:
The original code was trying to avoid yield with lock, as the lock is not released between each yield, which is bad for contention, specially as below we do a ToList, better return a List directly instead.